test/common/3scale_smtp.go

```diff
@@ -0,0 +1,490 @@
+		return secret.APIVersion, err
+	}
+
+	return "", nil
+}
+
+func patchSecret(ctx *TestingContext, t *testing.T) (string, error) {
+	// Update secret with our test smtp details
+	serviceIP, err := getServiceIP(ctx)
+
+	secret, err := getSecret(ctx)
+
+	if err != nil {
+		return "", err
+	}
+	originalHost = string(secret.Data["host"])
+	originalPassword = string(secret.Data["password"])
+	originalPort = string(secret.Data["port"])
+	originalUsername = string(secret.Data["username"])
+
+	secret = v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "redhat-rhmi-smtp",
+			Namespace: "redhat-rhmi-operator",
+		},
+		Data: map[string][]byte{},
+	}
+
+	secret.Data["host"] = []byte(serviceIP)
+	secret.Data["password"] = []byte(emailPassword)
+	secret.Data["port"] = []byte(emailPort)
+	secret.Data["username"] = []byte(emailUsername)
+
+	if err := ctx.Client.Update(goctx.TODO(), secret.DeepCopy(), &k8sclient.UpdateOptions{}); err != nil {
+		return secret.APIVersion, err
+	}
+
+	return "", nil
+}
+
+func getSecret(ctx *TestingContext) (v1.Secret, error) {
+
+	secret := &v1.Secret{}
+	if err := ctx.Client.Get(goctx.TODO(), types.NamespacedName{Name: "redhat-rhmi-smtp", Namespace: "redhat-rhmi-operator"}, secret); err != nil {
+		return *secret, err
+	}
+	return *secret, nil
+}
+
+func get3scaleSecret(ctx *TestingContext) (v1.Secret, error) {
+
+	secret := &v1.Secret{}
+	if err := ctx.Client.Get(goctx.TODO(), types.NamespacedName{Name: "system-smtp", Namespace: "redhat-rhmi-3scale"}, secret); err != nil {
+		return *secret, err
+	}
+	return *secret, nil
+}
+
+func getServiceIP(ctx *TestingContext) (string, error) {
+	service := &v1.Service{}
+
+	if err := ctx.Client.Get(goctx.TODO(), types.NamespacedName{Name: "smtp-server", Namespace: "smtp-server"}, service); err != nil {
+		return service.Spec.ClusterIP, err
+	}
+
+	return service.Spec.ClusterIP, nil
+}
+
+func createNamespace(ctx *TestingContext, t *testing.T) error {
+
+	// Create namespace for our test smtp-server
+	t.Log("Creating namespace, deployment and service")
+	nsSpec := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: testNamespace}}
+
+	_, err := ctx.KubeClient.CoreV1().Namespaces().Create(nsSpec)
+	if err != nil {
+		return fmt.Errorf("Unable to create namespace : %v", err)
+	}
+
+	// Create deployement for smtp-server
+	deployment := &k8sv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "smtp-server",
+		},
+		Spec: k8sv1.DeploymentSpec{
+			Replicas: &smtpReplicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": "smtp-server",
+				},
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": "smtp-server",
+					},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  "smtp-server",
+							Image: "quay.io/cathaloconnor/smtp-server:latest",
```

Comment:
@CathalOConnorRH Should we put the image into quay.io/integreatly?

Reply:
As discussed, this change will be made in a separate JIRA, when access to quay.io/integreatly/ is received. cc @CathalOConnorRH